src/psbt.cpp

```diff
@@ -125,17 +125,32 @@ uint256 PartiallySignedTransaction::GetUniqueID() const
     return mtx.GetHash();
 }
 
-bool PartiallySignedTransaction::AddInput(const CTxIn& txin, PSBTInput& psbtin)
+bool PartiallySignedTransaction::AddInput(PSBTInput& psbtin)
 {
-    if (std::find(tx->vin.begin(), tx->vin.end(), txin) != tx->vin.end()) {
+    if (std::find_if(inputs.begin(), inputs.end(),
+        [psbtin](const PSBTInput& psbt) {
```

Comment:
In https://github.com/bitcoin/bitcoin/commit/aeb9e44bc5a221553d2010752c76f8f16205551c "Change PSBT::AddInput to take just PSBTInput":

```
// Avoid duplicate input
[psbtin](const PSBTInput& psbt_input) {
```

(the comment about duplicate inputs is introduced in 3fe17931ae2d0e1b83cbe9c376f5d255ac9de68d "Implement PSBTv2 AddInput and AddOutput")

But this only works for v2, since `prev_txid` is nullptr otherwise. That's why the duplicate input check for v0 is maintained below.

It's more clear if you explicitly check the PSBT version in both places.

Reply:
Done